### PR TITLE
Utheque: URDF-theque (library of URDFs)

### PR DESCRIPTION
**Pull request by jbmouret**
**Objective:** make it easy to use the well-maintained URDF files without robot_dart (e.g., on a real robot, which does not have assimp, dart, etc.).

- We install the URDF files in `$PREFIX/share/utheque` instead of `$PREFIX/share/robot_dart` so that it makes more sense to use the same URDF if we do not care about robot_dart (e.g., for controllers)
- We can install the URDF even if robot_dart does not compile
- There is a small header-only library `utheque/utheque.hpp` that makes it easy to get a URDF from the library (e.g., in a controller)

TODO: option to deactivate explicitly robot_dart

### CODE REVIEW

File: cmake/UthequeConfigVersion.cmake.in [hunk 1]
```diff
@@ -0,0 +1,11 @@
+set(PACKAGE_VERSION "@robot_dart_VERSION@")
```
Comment (costashatz) [inline cmake/UthequeConfigVersion.cmake.in:1]: We keep robot_dart version here?

File: wscript [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env python
 # encoding: utf-8
+from gettext import install
```
Comment (costashatz) [inline wscript]: Why do we need this? I do not see this anywhere being used..

File: wscript [hunk 3]
```diff
@@ -42,9 +43,36 @@ def options(opt):
     opt.add_option('--shared', action='store_true', help='build shared library', dest='build_shared')
     opt.add_option('--tests', action='store_true', help='compile tests or not', dest='tests')
     opt.add_option('--python', action='store_true', help='compile python bindings', dest='pybind')
+    opt.add_option('--utheque-only', action='store_true', help='only install the URDF library (utheque) / deactivate RobotDART', dest='utheque_only')
+    opt.add_option('--no-robot_dart', action='store_true', help='only install the URDF library (utheque) / deactivate RobotDART', dest='utheque_only')
 
 
 def configure(conf):
+    if not conf.options.utheque_only:
+        robot_dart_ok = True
```
Comment (costashatz) [inline wscript]: `robot_dart_ok` seems not used anywhere..

File: wscript [hunk 4]
```diff
@@ -42,9 +43,36 @@ def options(opt):
     opt.add_option('--shared', action='store_true', help='build shared library', dest='build_shared')
     opt.add_option('--tests', action='store_true', help='compile tests or not', dest='tests')
     opt.add_option('--python', action='store_true', help='compile python bindings', dest='pybind')
+    opt.add_option('--utheque-only', action='store_true', help='only install the URDF library (utheque) / deactivate RobotDART', dest='utheque_only')
+    opt.add_option('--no-robot_dart', action='store_true', help='only install the URDF library (utheque) / deactivate RobotDART', dest='utheque_only')
```
Comment (costashatz) [inline wscript]: Why 2 options for the same thing?

**Review by costashatz — CHANGES_REQUESTED** [review]
Overall this is nice! See a few minor comments. Once they are done, I will format the code and merge! Thanks!

File: wscript (hunk 3 above)
Comment (jbmouret) [reply wscript]: Good catch. I moved to `env["...`

File: wscript (hunk 2 above)
Comment (jbmouret) [reply wscript]: I have no idea where this comes from!

File: cmake/UthequeConfigVersion.cmake.in (hunk 1 above)
Comment (jbmouret) [reply cmake/UthequeConfigVersion.cmake.in:1]: The alternative is to define a second version number, but do we want to keep things desynchronized?

Comment (costashatz) [reply cmake/UthequeConfigVersion.cmake.in:1]: Let's keep one.. Agreed..

File: wscript (hunk 2 above)
Comment (costashatz) [reply wscript]: Please remove it.. :)

File: wscript (hunk 4 above)
Comment (costashatz) [reply wscript:46]: I prefer having just one option... Keep the `--no-robot_dart`.

File: cmake/UthequeConfig.cmake.in [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+# CMAKE_MODULE_PATH
+set(CMAKE_MODULE_PATH "@Utheque_CMAKE_MODULE_PATH@")
+
+set(Utheque_INCLUDE_DIRS "@Utheque_INCLUDE_DIRS@")
+
+
+find_package(Boost REQUIRED filesystem)
+
+set(Utheque_LIBRARIES "Threads::Threads;Eigen3::Eigen;Boost::filesystem")
```
Comment (costashatz) [inline cmake/UthequeConfig.cmake.in:18]: Do we need `pthread` and `Eigen` for utheque? I would say no.. Just `boost::filesystem`.

Comment (costashatz) [reply cmake/UthequeConfig.cmake.in]: @jbmouret I already removed those.. Let me know if you think this is an issue..

**Review by costashatz — APPROVED** [review]

File: cmake/UthequeConfig.cmake.in (hunk 5 above)
Comment (jbmouret) [reply cmake/UthequeConfig.cmake.in]: No problem.

**Comment by jbmouret** [comment]
Please note that I did not try the CMake integration yet.

**Comment by costashatz** [comment]
@jbmouret CMake integration should be working now. I tried on my PC and added a test in the CI. Check also in your Mac. If it works there as well, we should be ready to merge...

**Pull request merged**